pilot/pkg/networking/core/v1alpha3/name_table.go

```diff
@@ -74,12 +74,9 @@ func (configgen *ConfigGeneratorImpl) BuildNameTable(node *model.Proxy, push *mo
 		// IP allocation logic for service entry was unable to allocate an IP.
 		if svcAddress == constants.UnspecifiedIP {
 			// For all k8s headless services, populate the dns table with the endpoint IPs as k8s does.
-			// And for each individual pod, populate the dns table with the endpoint IP with a manufactured host name from
-			// "statefulset.kubernetes.io/pod-name" that k8s sets.
+			// And for each individual pod, populate the dns table with the endpoint IP with a manufactured host name.
 			if svc.Attributes.ServiceRegistry == string(serviceregistry.Kubernetes) &&
 				svc.Resolution == model.Passthrough && len(svc.Ports) > 0 {
-				// TODO: this is used in two places now. Needs to be cached as part of the headless service
-				// object to avoid the costly lookup in the registry code
 				for _, instance := range push.ServiceInstancesByPort(svc, svc.Ports[0].Port, nil) {
```

Comment:
This TODO is not relevant because if we pass labels nil, we do not look up discovery